processing/src/main/java/io/druid/query/extraction/MatchingDimExtractionFn.java

```diff
@@ -59,9 +60,8 @@ public MatchingDimExtractionFn(
   @Override
   public String apply(String dimValue)
   {
-    dimValue = (dimValue == null) ? "" : dimValue;
-    Matcher matcher = pattern.matcher(dimValue);
-    return matcher.find() ? dimValue : null;
+    Matcher matcher = pattern.matcher(Strings.nullToEmpty(dimValue));
```

Comment:
if `dimValue` is null, then it doesn't matter whether the pattern matches or not, we will always return null, so we can skip both the nullToEmpty and the pattern matching, no?